tunnel/intercept/svcpoll.go

```diff
@@ -234,3 +257,104 @@ func (self *ServiceListener) host(svc *entities.Service) {
 		}
 	}
 }
+
+func (self *ServiceListener) configureSourceAddrProvider(svc *entities.Service) error {
+	if svc.InterceptV1Config == nil || svc.InterceptV1Config.SourceIp == nil || *svc.InterceptV1Config.SourceIp == "" {
+		return nil
+	}
+
+	sourceIp := *svc.InterceptV1Config.SourceIp
+
+	if sourceIp == sourceAddrTemplKey {
+		svc.SourceAddrProvider = func(sourceAddr, _ net.Addr) string {
+			return sourceAddr.String()
+		}
+		return nil
+	}
+
+	if sourceIp == destAddrTemplKey {
+		svc.SourceAddrProvider = func(_, destAddr net.Addr) string {
+			return destAddr.String()
+		}
+		return nil
+	}
+
+	currentIdentity, err := self.provider.GetCurrentIdentity()
+	if err != nil {
+		return err
+	}
+
+	sourceIp = strings.ReplaceAll(sourceIp, "$tunneler_id.name", currentIdentity.Name)
+	if sourceIp, err = self.replaceIdTags(sourceIp, currentIdentity); err != nil {
+		return err
+	}
+
+	if strings.IndexByte(sourceIp, '$') < 0 {
+		svc.SourceAddrProvider = func(_, _ net.Addr) string {
+			return sourceIp
+		}
+		return nil
+	}
+
+	svc.SourceAddrProvider = func(sourceAddr, destAddr net.Addr) string {
+		sourceAddrIp, sourceAddrPort := tunnel.GetIpAndPort(sourceAddr.String())
+		destAddrIp, destAddrPort := tunnel.GetIpAndPort(destAddr.String())
+		result := strings.ReplaceAll(sourceIp, sourceIpTemplKey, sourceAddrIp)
+		result = strings.ReplaceAll(result, sourcePortTemplKey, sourceAddrPort)
+		result = strings.ReplaceAll(result, destIpTemplKey, destAddrIp)
+		result = strings.ReplaceAll(result, destPortTemplKey, destAddrPort)
+		return result
+	}
+
+	return nil
+}
+
```

Comment:
@scareything note I changed this from tags to appData. Let me know if that's ok or if we want to keep it tags and just have it use appData under the hood